lib/model/unittest/CMetricModelTest.cc

```diff
@@ -57,176 +57,35 @@ namespace {
-    CDataGatherer::TStrCPtrVec fieldValues;
-    fieldValues.push_back(&person);
-    if (inf1) {
-        fieldValues.push_back(&(inf1.get()));
-    }
-    if (inf2) {
-        fieldValues.push_back(&(inf2.get()));
-    }
-    std::string valueAsString;
-    valueAsString += core::CStringUtils::typeToStringPrecise(lat, core::CIEEE754::E_DoublePrecision);
-    valueAsString += model::CAnomalyDetectorModelConfig::DEFAULT_MULTIVARIATE_COMPONENT_DELIMITER;
-    valueAsString += core::CStringUtils::typeToStringPrecise(lng, core::CIEEE754::E_DoublePrecision);
-    fieldValues.push_back(&valueAsString);
-
-    CEventData eventData;
-    eventData.time(time);
-
-    gatherer.addArrival(fieldValues, eventData, resourceMonitor);
-}
-
-CEventData makeEventData(core_t::TTime time,
-                         std::size_t pid,
-                         double value,
-                         const TOptionalStr& influence = TOptionalStr()) {
-    CEventData result;
-    result.time(time);
-    result.person(pid);
-    result.addAttribute(std::size_t(0));
-    result.addValue({value});
-    result.addInfluence(influence);
-    return result;
-}
-
-TDouble1Vec featureData(const CMetricModel& model,
-                        model_t::EFeature feature,
-                        std::size_t pid,
-                        core_t::TTime time) {
-    const CMetricModel::TFeatureData* data = model.featureData(feature, pid, time);
-    if (!data) {
-        return TDouble1Vec();
-    }
-    return data->s_BucketValue ? data->s_BucketValue->value() : TDouble1Vec();
+const CModelTestFixtureBase::TSizeDoublePr1Vec NO_CORRELATES;
 }
 
-TDouble1Vec multivariateFeatureData(const CMetricModel& model,
-                                    model_t::EFeature feature,
-                                    std::size_t pid,
-                                    core_t::TTime time) {
-    const CMetricModel::TFeatureData* data = model.featureData(feature, pid, time);
-    if (!data) {
-        return TDouble1Vec();
-    }
-    return data->s_BucketValue ? data->s_BucketValue->value() : TDouble1Vec();
-}
-
-void processBucket(core_t::TTime time,
-                   core_t::TTime bucketLength,
-                   const TDoubleVec& bucket,
-                   const TStrVec& influencerValues,
-                   CDataGatherer& gatherer,
-                   CResourceMonitor& resourceMonitor,
-                   CMetricModel& model,
-                   SAnnotatedProbability& probability) {
-    for (std::size_t i = 0u; i < bucket.size(); ++i) {
-        addArrival(gatherer, resourceMonitor, time, "p", bucket[i],
-                   TOptionalStr(influencerValues[i]));
+class CTestFixture : public CModelTestFixtureBase {
+public:
+    CEventData makeEventData(core_t::TTime time,
+                             std::size_t pid,
+                             double value,
+                             const TOptionalStr& influence = TOptionalStr()) {
```

Comment:
I think it should be possible to move this into the base class. We have `makeEventData` in `CEventRateModelTest` which I think this could be combined with.